src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ITypeSymbolExtensions.TypeSyntaxGeneratorVisitor.cs

```diff
@@ -127,18 +127,56 @@ public static bool TryCreateNativeIntegerType(INamedTypeSymbol symbol, out TypeS
 
 #if !CODE_STYLE
 
+#nullable enable
             public override TypeSyntax VisitFunctionPointerType(IFunctionPointerTypeSymbol symbol)
             {
-                // TODO(https://github.com/dotnet/roslyn/issues/39865): generate the calling convention once exposed through the API
+                FunctionPointerCallingConventionSyntax? callingConventionSyntax = null;
+                if (symbol.Signature.CallingConvention != System.Reflection.Metadata.SignatureCallingConvention.Default)
+                {
+                    SeparatedSyntaxList<FunctionPointerUnmanagedCallingConventionSyntax> conventionsList = default;
+                    switch (symbol.Signature.CallingConvention)
+                    {
+                        case System.Reflection.Metadata.SignatureCallingConvention.CDecl:
+                            conventionsList = SyntaxFactory.SeparatedList(new[] { GetConventionForString("Cdecl") });
+                            break;
+                        case System.Reflection.Metadata.SignatureCallingConvention.StdCall:
+                            conventionsList = SyntaxFactory.SeparatedList(new[] { GetConventionForString("Stdcall") });
+                            break;
+                        case System.Reflection.Metadata.SignatureCallingConvention.ThisCall:
+                            conventionsList = SyntaxFactory.SeparatedList(new[] { GetConventionForString("Thiscall") });
+                            break;
+                        case System.Reflection.Metadata.SignatureCallingConvention.FastCall:
+                            conventionsList = SyntaxFactory.SeparatedList(new[] { GetConventionForString("Fastcall") });
+                            break;
+
+                        default:
+                            // All types that come from CallingConventionTypes start with "CallConv". We don't want the prefix for the actual
+                            // syntax, so strip it off
```

Comment:
What happens for the broken code case? I saw you had a test -- is the compiler synthesizing an error type that has a CallConv prefix in that case?

Reply:
Yep.

---
In reply to: [463854656](https://github.com/dotnet/roslyn/pull/46467#discussion_r463854656) [](ancestors = 463854656)

Reply:
Or, for a missing identifier, you never see it in the first place.

---
In reply to: [463863287](https://github.com/dotnet/roslyn/pull/46467#discussion_r463863287) [](ancestors = 463863287,463854656)